opencog/atoms/pattern/PatternLink.cc

```diff
@@ -889,7 +890,9 @@ void PatternLink::make_term_tree_recursive(const Handle& root,
 	_pat.connected_terms_map[{h, root}].emplace_back(ptm);
 
 	// Update the PatternTerm quotation level
-	if (QUOTE_LINK == t)
+	if (not parent->isQuoted() and LOCAL_QUOTE_LINK == t)
+		ptm->setLocalQuote(true);
```

Comment:
Actually there's might be a simpler way indeed! Consider that negative quotation level encodes alocal quotation. This may work because a `LocalQuoteLink` is only active if it is so far unquoted, as explained here http://wiki.opencog.org/w/LocalQuoteLink.